io/subsystem/src/main/java/org/wildfly/extension/io/IOSubsystemRegistrar.java

```diff
@@ -52,10 +50,8 @@ class IOSubsystemRegistrar implements SubsystemResourceDefinitionRegistrar, Reso
 
     static final ModelNode LEGACY_DEFAULT_WORKER = new ModelNode("default");
 
-    static final AttributeDefinition DEFAULT_WORKER = new SimpleAttributeDefinitionBuilder("default-worker", ModelType.STRING)
+    static final CapabilityReferenceAttributeDefinition<XnioWorker> DEFAULT_WORKER = new CapabilityReferenceAttributeDefinition.Builder<>("default-worker", CapabilityReference.builder(DEFAULT_WORKER_CAPABILITY, IOServiceDescriptor.WORKER).build())
```

Comment:
@pferraro Shouldn't there be more to this, i.e. a change to IOSubsystemRegistar.configure?

Reply:
Hmm - indeed.  I had added a ServiceDependency.isEmpty() method to better handle the optional capability reference case, but it looks like I stashed that work without committing it.  Will fix.

Reply:
Fixed.